napari/layers/labels/labels.py

```diff
@@ -525,7 +537,37 @@ def paint(self, coord, new_label, refresh=True):
             slice_coord = tuple(slice_coord)
 
         # update the labels image
-        self.data[slice_coord] = new_label
+
+        if self._overwrite:
+            self.data[slice_coord] = new_label
+        else:
+            self.paint_without_overwrite(self.data, slice_coord, new_label, 0)
 
         if refresh is True:
             self.refresh()
+
+    def paint_without_overwrite(
+        self, array, slice_coord, new_label, dimension
```

Comment:
when `new_label == 0` ... painting with overwrite `False` never does anything right?  I can't think of anything more interesting at the moment... but is there anything we could use this for in that scenario?

Reply:
that is correct, it doesn't really matter what the target label is if the user want to avoid write over already labeled data. However, it is a good point since we can repurpose it to be a "eraser" function, which is raised in the same issue. We need more UXR input to map out if the behavior is desired